src/entity/Entity.php

```diff
@@ -658,6 +658,18 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 			$ev = new EntityDamageEvent($this, EntityDamageEvent::CAUSE_VOID, 10);
 			$this->attack($ev);
 			$hasUpdate = true;
+		}else{
+			$entityBlock = $this->getWorld()->getBlock($this->getLocation());
+			$entityBox = $this->getBoundingBox()->expandedCopy(0.001, 0.1, 0.001);
```

Comment:
Initially, I started testing with 0.001 for each axis but I found a bug on the Y axis with a slab. Indeed, if you put a block of magma under a slab and a player steps on this slab, with 0.001 on the Y axis, the collision is not detected so I put 0.1 to detect it

Reply:
That doesn't make any sense. Is magma supposed to hurt you from underneath a slab? And if it is, why does 0.1 make a difference since the BB of a slab is 0.5 thick?

Reply:
After some reseach, player take damage on magma block if block between player and magma block has AABB below or equal to 13/16 like trapdoor (https://minecraft.wiki/w/Magma_Block#Damage). So the y value is 1-13/16 = 3/16.